db/models/articlesModel.js

```diff
@@ -11,14 +11,19 @@ exports.selectArticleById = (articleId) => {
     });
 };
 
-exports.selectArticles = (sort_by = "created_at") => {
+exports.selectArticles = (topic) => {
+  const param = [];
   let queryString = `SELECT articles.author, articles.title, articles.article_id, articles.topic, 
   articles.created_at, articles.votes, articles.article_img_url, COUNT (comments.comment_id):: int AS comment_count
-  FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id 
-  GROUP BY  articles.author, articles.title, articles.article_id, articles.topic, 
+  FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id `;
+  if (topic) {
+    queryString += `WHERE articles.topic = $1 `;
+    param.push(topic);
+  }
+  queryString += `GROUP BY articles.author, articles.title, articles.article_id, articles.topic, 
   articles.created_at, articles.votes, articles.article_img_url
-  ORDER BY articles.${sort_by} DESC;`;
-  return db.query(queryString).then((data) => {
+  ORDER BY articles.created_at DESC;`;
+  return db.query(queryString, param).then((data) => {
     return data.rows;
   });
 };
```

Comment:
This is a nice construction of your query string and params array - It's good that you changed how your sort_by is working as well, as this would have been susceptible to SQL injection